### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jackparmer**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,3 +21,4 @@ requests==2.28.1
 # Testing
 pytest==7.1.3
 black==23.3.0
+ticlib
```
Comment (itsjoeoui) [inline requirements.txt:24]: We should probably pin this as well!

File: src/feature/common/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      );
+    }
+
+    var content;
+
+    if (textInput !== "") {
+      //case 1: name is included in the string of the section node or leaf node
+      if (node.title.toLowerCase().includes(textInput.toLocaleLowerCase())) {
+        //case 1.1: node has children (is a section)
+        if (node["child"] != null && !("key" in node)) {
```
Comment (itsjoeoui) [inline src/feature/common/Sidebar/Sidebar.tsx:117]: Looks like this should be `!==` instead? @izi-on

File: src/feature/common/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  sections: Node;
+  leafNodeClickHandler: leafClickHandler;
+  manifestMap: any; //Key value pair object
+}) => {
+  const [textInput, handleChangeInput] = useState("");
+  const addButtonClass = useAddButtonStyle();
+  const { classes } = useSidebarStyles();
+
+  //this function will create the sections to be rendered according to the search input
+  const renderSection = (textInput: string, node: Node, depth: number) => {
```
Comment (itsjoeoui) [inline src/feature/common/Sidebar/Sidebar.tsx:102]: @izi-on Can you also add the return type of this function please

File: src/feature/common/Sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+              />
+            );
+          }
+        }
+      }
+
+      //case 3: no search input
+    } else {
+      //case 3.1: node has children (is a section)
+      if (node["child"] != null && !("key" in node)) {
```
Comment (itsjoeoui) [inline src/feature/common/Sidebar/Sidebar.tsx:171]: Should be `!==`, there are a couple `==` below that should be changed as well

File: src/feature/common/defaultPlotLayout.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Layout } from "plotly.js";
 
 const styledPlotLayout = (theme: "light" | "dark") => {
-  const plotFeatureColor = "#282c34";
+  const plotFeatureColor = theme === "light" ? "#282c34" : "#fff";
```
Comment (itsjoeoui) [inline src/feature/common/defaultPlotLayout.ts:4]: We may want to use the theme we defined instead of hard-coding

**Review by itsjoeoui — CHANGES_REQUESTED** [review]

**Review by itsjoeoui — DISMISSED** [review]

File: requirements.txt [hunk 6]
```diff
@@ -13,12 +13,11 @@ PyYAML==6.0
...
+uptime==3.0.1
+schedule==1.2.0
 djangorestframework==3.14.0
 python-dotenv==0.21.0
 requests==2.28.1
-# Testing
 pytest==7.1.3
 black==23.3.0
-ticlib
+ticlib==0.2.2
```
Comment (smahmed776) [inline requirements.txt:23]: This is a node-specific dependency and shouldn't be here.

**Review by smahmed776 — CHANGES_REQUESTED** [review]

**Review by smahmed776 — APPROVED** [review]

**Pull request merged**